backend/crawler/crawler.py

```diff
@@ -153,43 +219,71 @@ def parse(year, semester, lectures):
         try:
             department_instance = Department.objects.get(name=lecture['department'])
         except Exception as e:
```

Comment:
Use `ObjectDoesNotExist` rather than just `Exception`
It is difficult to get the reason of using try except.
Also same to L228, L240